src/coreclr/pal/src/thread/thread.cpp

```diff
@@ -1581,6 +1586,34 @@ GetThreadTimes(
     return (retval);
 }
 
+HRESULT
+PALAPI
+GetThreadDescription(
```

Comment:
I think we can delete this function from here and implement it a bit differently in palsuite due to the following reasons:
* some platforms like Alpine Linux and FreeBSD do not have `pthread_getname_np` (in the versions we are using in CI)
* this API is not used by the product.

then in palsuite, (for now) we can:
* use `pthread_self()` call `pthread_getname_np()` on it.
* execute tests only for `defined(_APPLE_) || (defined(__linux__) && !defined(TARGET_ALPINE_LINUX))`

Reply:
I had originally tried that (using `pthread_getname_np()` in the test to read the thread name) but couldn't get that to compile because it looks like there's some type collisions with typedefs between PAL and the MacOS SDK. Idk if there's a good way to get around this or not. I get this when trying to include `<palsuite.h>` and `<pthread.h>`
<img width="1311" alt="Screen Shot 2021-01-27 at 2 38 23 PM" src="https://user-images.githubusercontent.com/675417/106044258-78f71700-60ad-11eb-84bb-32b1eca2feee.png">

Reply:
Ah true; the include combination of PAL is bit tricky. We just need to separate the implementation in a separate source file: https://github.com/am11/runtime/commit/40d7d853b6bfd3472a6c223c55d9e1df7002f3f8 on top of your branch. Tested on Linux and macOS with:

```bash
$ ./build.sh clr.paltests
$ ./src/coreclr/pal/tests/palsuite/runpaltests.sh $(pwd)/artifacts/bin/coreclr/OSX.x64.Debug/paltests
# replace OSX.x64.Debug with Linux.x64.Debug for Linux

... snip ...

PAL Test Results:
  Passed: 698
  Failed: 0
```

Reply:
I am not sure sure whether this test is worth all this trouble. It does not test whether the managed API (what we actually care about) works.

Reply:
I think all ~700 PAL tests are no different; testing PAL API surface rather ensuring managed API?

Reply:
It looks like `pthread_getname_np()` isn't implemented in musl (and FreeBSD) so builds that rely on that are still failing; the tests are still building even though they aren't added to the test suite. I couldn't find any `#defines` that we can use for `GetThreadName()` so at this point it may be easier just to remove the tests, but I'm open to any other ideas :)

Reply:
> I think all ~700 PAL tests are no different; testing PAL API surface rather ensuring managed API works?

Most of the PAL tests are testing complex APIs that are not directly exposed as managed API, and the verification is focused on making sure that the API has identical behavior as its Windows equivalent.

Reply:
Yup. On Windows and Linux, Thread.Name_set() updates the name internal to runtime as well as system-wide. On macOS it was only updating the internal name which PR has fixed. Thread.Name_get() returns the internal value on all platforms. This test is ensuring that the name change due to Thread.Name_set() call is reflected on system level, observable by external tools.

Since there is no managed API change, perhaps this system-wide change test is something we don't want to cover. It doesn't hurt IMO. :)